File: apps/party-time-backend/src/main/java/com/partytime/jpa/repository/EventRepository.java
package com.partytime.jpa.repository;

import com.partytime.jpa.entity.Event;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface EventRepository extends JpaRepository<Event, Long> {
    boolean existsByName(String name);
    List<Event> findByOrganizer_Email(String email);



}

File: apps/party-time-backend-kotlin/src/main/kotlin/com/partytime/jpa/repository/EventRepository.kt
package com.partytime.jpa.repository

import com.partytime.jpa.entity.Event
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface EventRepository: JpaRepository<Event, Long> {
    fun existsByName(name: String?): Boolean
    fun findByOrganizer_Email(email: String): List<Event>
}
